python/kserve/kserve/storage/storage.py

```diff
@@ -284,8 +283,26 @@ def _download_s3(uri, temp_dir: str) -> str:
                 temp_dir = Storage._unpack_archive_file(target, mimetype, temp_dir)
         return temp_dir
 
+    @staticmethod
+    def _download_hf(uri, temp_dir: str) -> str:
+        from huggingface_hub import snapshot_download
+
+        components = uri.split("://")[1].split("/")
+
+        repo = components[0]
+        model, _, hash_value = components[1].partition(":")
+        revision = hash_value if hash_value else None
+
+        snapshot_download(
+            repo_id=f"{repo}/{model}", revision=revision, local_dir=temp_dir
```

Comment:
HF repo usually contains multiple formats, for example, `safetensors`, `bin` etc. Most of the case, only one should be downloaded to reduce the download time (should save half of the time). 

Usually I will add the allow/ignore patterns, for example:

```python
# here safetensors is the preferred format.
allow_patterns = ["*.json", "*.safetensors", "*.model"]
ignore_patterns = ["*.msgpack", "*.h5", "*.bin"]

output_dir = snapshot_download(
      repo_id=model_name,
      revision=revision,
      allow_patterns=allow_patterns,
      ignore_patterns=ignore_patterns,
      local_dir=models_path,
)
```
I think it is quite needed, but I am not sure the best way to pass it, for example via the `ALLOW_PATTERNS` and `IGNORE_PATTERNS` in the secret?